library/pk.c

```diff
@@ -850,6 +879,100 @@ mbedtls_pk_type_t mbedtls_pk_get_type(const mbedtls_pk_context *ctx)
 }
 
 #if defined(MBEDTLS_USE_PSA_CRYPTO)
+#if defined(MBEDTLS_ECP_C)
+int mbedtls_pk_get_public_key(mbedtls_pk_context *pk, unsigned char *buf,
+                              size_t buf_size, size_t *key_len)
+{
+    if ((pk == NULL) || (pk->pk_raw_len == 0)) {
+        return MBEDTLS_ERR_PK_BAD_INPUT_DATA;
+    }
+    if (buf_size < MBEDTLS_PK_MAX_EC_PUBKEY_RAW_LEN) {
```

Comment:
```suggestion
    if (buf_size < pk->pk_raw_len) {
```

Reply:
I also fixed the `memcpy` few lines below with the same solution:
`MBEDTLS_PK_MAX_EC_PUBKEY_RAW_LEN` --> `pk->pk_raw_len`